### DISCUSSION TITLE
Slow ADC on ESP32

### DISCUSSION THREAD

**Discussion by tosmaz** [micropython/micropython#10561] [ESP32]
Why does simple ADC.read() take ~300uS? It seems very slow if we are speaking about 200kSps ADC. Is there any hack to make it faster?

**Comment by bixb922** [comment]
With @micropython.native and this code, I get about 43 microseconds per sample
```py
    read = adc.read_u16
    for i in range(len(signal)):
        signal[i] = read()
```

**Comment by bixb922** [comment]
With machine.freq(240_000_000) and @micropython.native time gets a bit lowe

**Comment by jimmo** [comment]
On stm32 we have `adc.read_timed` and `adc.read_timed_multi`, which might be worth implementing on other ports like ESP32. https://docs.micropython.org/en/latest/library/pyb.ADC.html?highlight=read_timed#pyb.ADC.read_timed

**Reply by robert-hh** [reply]
This is implemented in a PR for SAMD as well, but still in the queue, maybe a loooong time.
